packages/react/src/components/DataTableSkeleton/DataTableSkeleton.tsx

```diff
@@ -10,6 +10,49 @@ import React from 'react';
 import cx from 'classnames';
 import { usePrefix } from '../../internal/usePrefix';
 
+export interface DataTableSkeletonProps {
+  /**
+   * Specify an optional className to add.
+   */
+  className?: string;
+
+  /**
+   * Specify the number of columns that you want to render in the skeleton state
+   */
+  columnCount: number;
+
+  /**
+   * Optionally specify whether you want the Skeleton to be rendered as a
+   * compact DataTable
+   */
+  compact: boolean;
+
+  /**
+   * Optionally specify the displayed headers
+   */
+  headers?: [{ header: string, key: string }] | { header: string, key: string };
```

Comment:
The pre-existing header type is an array or an object with a key. The object with a key, I do not think is being used. However, to remove it would break the API and require a major version bump. The array itself expects an object with a header.

```
  headers: PropTypes.oneOfType([
    PropTypes.array,
    PropTypes.shape({
      key: PropTypes.string,
    }),
  ]),
  ```